### PR TITLE
Feat- Add to cart

### PR DESCRIPTION
**Pull request by abhij1607**
- Add to cart functionality
- increase or decrease cart products on the cart page
- move to wishlist

Please ignore abstractions, for now, will do it later. Also, ignore March 31st commits, already reviewed in previous PR

### CODE REVIEW

File: src/pages/Wishlist/wishlist.js [hunk 1]
```diff
@@ -1,143 +1,54 @@
...
     <div className="wrapper">
       <main className="main-pane auto-container">
         <h1 className="h2">
           My Wishlist
-          <span className="p-lg txt-wt-light">6 Items</span>
+          <span className="p-lg txt-wt-light">
+            ` {productState.wishlist.length} Items`
+          </span>
         </h1>
         {/* category list cards */}
```
Comment (saadahmadhere) [inline src/pages/Wishlist/wishlist.js]: The section name says it all, no need for comments

Comment (abhij1607) [reply src/pages/Wishlist/wishlist.js]: yep, thanks for pointing it out

**Review by saadahmadhere — COMMENTED** [review]
Nice & neat. Good use of 1 useContext for both cart and wish list functionalities.

File: src/context/user-products-context.jsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+          (product) => product._id !== action.payload._id
+        ),
+      };
+
+    case "ADD_TO_CART":
+      let itemFound = false;
+      const updatedCart = [...state.cart].map((item) => {
+        if (item._id === action.payload._id) {
+          itemFound = true;
+          console.log(item);
```
Comment (Tanay-J) [inline src/context/user-products-context.jsx]: this can be removed

**Review by Tanay-J — APPROVED** [review]
Good work! Code is clean and easy on eyes. Got to see how can wishlist and cart management be done using single context.

Suggestions:
- alt texts can be improved
- to make the code even more modular, you may extract reducer functions into separate files

File: src/context/user-products-context.jsx (hunk 2 above)
Comment (abhij1607) [reply src/context/user-products-context.jsx]: sure, thanks for pointing it out

**Pull request merged**